presto-main/src/main/java/io/prestosql/server/protocol/Query.java

```diff
@@ -575,8 +576,13 @@ private URI createPartialCancelUri(int stage, UriInfo uriInfo, long nextToken)
 
     private Column createColumn(String name, Type type)
     {
-        TypeSignature signature = type.getTypeSignature();
-        return new Column(name, type.getDisplayName(), toClientTypeSignature(signature));
+        String typeName = type.getDisplayName();
+
+        if (type instanceof TimestampType && !supportsParametricDateTime) {
```

Comment:
Should we compare the type signature so that it's consistent with `toClientTypeSignature`?

Reply:
Actually, I realized this has a problem. It works for top-level timestamps, but not for, say, `array(timestamp)`

Reply:
@martint add this as desired test-cases in https://github.com/prestosql/presto/issues/1787